File: frontend/src/types/activityEvent.ts
import type { LongId } from '@/types/common'

/** 字段变更 */
export interface FieldChange {
  fieldName: string
  oldValue: string | null
  newValue: string | null
}

/** 活动事件 */
export interface ActivityEvent {
  id: string
  requestId: string | null

  resourceType: string
  resourceId: string
  resourceName: string | null

  action: string
  actionLabel: string | null
  result: 'SUCCESS' | 'FAILURE' | 'PARTIAL'
  errorMessage: string | null

  changedFields: FieldChange[] | null
  beforeSnapshot: string | null
  afterSnapshot: string | null

  userId: LongId | null
  userName: string | null

  sourceIp: string | null
  userAgent: string | null
  apiEndpoint: string | null
  httpMethod: string | null

  reason: string | null
  module: string | null
  tags: Record<string, string> | null

  occurredAt: string
}

/** 查询参数 */
export interface ActivityEventQuery {
  module?: string
  resourceType?: string
  resourceId?: string
  action?: string
  userId?: LongId
  result?: string
  startTime?: string
  endTime?: string
  keyword?: string
  pageNum?: number
  pageSize?: number
}

/** 统计数据 */
export interface ActivityEventStats {
  total: number
  success: number
  failed: number
  today: number
}

File: backend/src/main/java/com/school/management/infrastructure/activity/ActivityEvent.java
package com.school.management.infrastructure.activity;

import com.school.management.domain.shared.model.valueobject.FieldChange;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;

/**
 * 统一活动事件（不可变 POJO）
 */
@Getter
@Builder
public class ActivityEvent {

    private final String requestId;

    // 资源标识
    private final String resourceType;
    private final String resourceId;
    private final String resourceName;

    // 动作
    private final String action;
    private final String actionLabel;
    @Builder.Default
    private final String result = "SUCCESS";
    private final String errorMessage;

    // 变更追踪
    private final List<FieldChange> changedFields;
    private final String beforeSnapshot;
    private final String afterSnapshot;

    // 操作人
    private final Long userId;
    private final String userName;

    // HTTP 上下文
    private final String sourceIp;
    private final String userAgent;
    private final String apiEndpoint;
    private final String httpMethod;

    // 元数据
    private final String reason;
    private final String module;
    private final Map<String, String> tags;

    // 时间
    @Builder.Default
    private final LocalDateTime occurredAt = LocalDateTime.now();
}
